src/main/java/org/dependencytrack/resources/v1/BomResource.java

```diff
@@ -226,6 +227,8 @@ public Response uploadBom(BomSubmitRequest request) {
                         project = qm.createProject(StringUtils.trimToNull(request.getProjectName()), null, StringUtils.trimToNull(request.getProjectVersion()), null, null, null, true, true);
                         Principal principal = getPrincipal();
                         qm.updateNewProjectACL(project, principal);
+                        final ProjectCreationEvent projectCreationEvent = new ProjectCreationEvent(project.getUuid(), project.getName());
+                        Event.dispatch(projectCreationEvent);
```

Comment:
I did it this way because that's how I saw it done for a VulnerabilityAnalysisEvent in FindingResource, f.ex. 
Is there a reason to do it differently there?

Reply:
An event would be used if there's a need for some action to be taken (e.g. performing some sort of analysis). 

> I did it this way because that's how I saw it done for a VulnerabilityAnalysisEvent in FindingResource and the BomUploadEvent in the same file, f.ex.

Both of these events have `Subscriber`s that listen for them to take action. `VulnerabilityAnalysisEvent` will cause a vulnerability analysis to occur, `BomUploadEvent` triggers a `BomUploadProcessingTask` which is responsible for, well, processing the BOM. These processes may or may not dispatch notifications as well, but that is not their main purpose.

Internally, events and notifications work in the same way: There is a queue with one or more subscribers. Subscribers are executed asynchronously in dedicated thread pools. DT uses separate queues for events and notifications.

If your only intention (as of now) is to publish a notification, then there's no reason to do it by publishing an event first. I agree that at some point there may be other areas of DT that may want to react to new projects being created, but we can still refactor accordingly if we reach that point. For the time being I'd prefer to limit the amount of code we have to maintain.